### PR TITLE
Add support for layer exclusions

### LINKED ISSUE #1345: Exclusions des couches
Cela va de soi, mais @jaykayone, peux-tu confirmer que nous allons devoir gérer les exclusions de couches ?

### PR DESCRIPTION
**Pull request by pgiraud**
Fixes #1345

### CODE REVIEW

**Comment by tonio** [comment]
Shouldn't we listen to layer’s removal to update exclusions list? (I guess addition can’t be checked if `addLayer` is used instead of `addLayerById`)

**Comment by tonio** [comment]
One question, otherwise LGTM.

**Comment by pgiraud** [comment]
Pull request updated. @tonio can you please have a look?

File: jsapi/src/main.js [hunk 1]
```diff
@@ -285,7 +295,71 @@ lux.Map.prototype.addLayers_ = function(layers) {
...
+  var concat = arr1.concat(arr2);
+  goog.array.sort(concat);
+  var i;
+  var len = concat.length;
+  for (i = 1; i < len; i++) {
+    if (concat[i] == concat[i - 1]) {
+      return true;
+    }
+  }
+  return false;
```
Comment (tonio) [inline jsapi/src/main.js]: I may miss something but couldn’t `L349:L358` be replaced by?
```javascript
return arr1.filter(v => arr2.indexOf(v) > -1).length > 0
```

**Comment by tonio** [comment]
LGTM

**Pull request merged**